openquake/hazardlib/gsim/yenier_atkinson_2015.py

```diff
@@ -0,0 +1,559 @@
+                if imt == PGA():
+                    # avoid recomputing PGA on rock, just add site terms
+                    mean = pga4nl + \
+                        self._get_site_amplification_linear(sites.vs30, C) + \
+                        self._get_site_amplification_non_linear(sites.vs30, pga4nl, C)
+                else:
+                    mean = self._compute_distance_scaling(rup, dists, C) + \
+                        self._compute_magnitude_scaling(rup, C) + \
+                        self._compute_stress_adjustment(rup, C_SA) + \
+                        self._get_site_amplification_linear(sites.vs30, C) + \
+                        self._get_site_amplification_non_linear(sites.vs30, pga4nl, C)
+
+                stddevs = self._get_stddevs(C, stddev_types, num_sites=len(sites.vs30))
+        
+                return mean, stddevs
+        
+        def _get_stddevs(self, C, stddev_types, num_sites):
+                """
+                Return standard deviations as defined in table 8, pag 121.
+                """
+                stddevs = []
+                for stddev_type in stddev_types:
+                    assert stddev_type in self.DEFINED_FOR_STANDARD_DEVIATION_TYPES
+                    if stddev_type == const.StdDev.TOTAL:
+                        stddevs.append(C['std'] + np.zeros(num_sites))
+                    elif stddev_type == const.StdDev.INTRA_EVENT:
+                        stddevs.append(C['sigma'] + np.zeros(num_sites))
+                    elif stddev_type == const.StdDev.INTER_EVENT:
+                        stddevs.append(C['tau'] + np.zeros(num_sites))
+                return stddevs
+    
+        def _compute_distance_scaling(self, rup, dists, C):
+                """
+                Compute distance scaling term, equations (5), (6), (7), and (8), pag 1991.
+                """
+                # Computing geometrical spreading, Fz.
+                h = 10 ** (-0.405 + 0.235 * rup.mag)
+                R = np.sqrt(np.amin(dists.rrup) ** 2 + (h ** 2))
+                Rref = np.sqrt(1 + (h ** 2))
+                b1 = -1.3
+                b2 = -0.5
+                Rt = 50		# Transition distance of 50km.
+                if R <= Rt:
+                        Z = R ** b1
+                else:
+                        Z = (Rt ** b1) * ((R/Rt) ** b2)
+                return np.log(Z) + ((C['b3'] + (C['b4'] * rup.mag)) * np.log(R / Rref))
+        
+        def _compute_magnitude_scaling(self, rup, C):
+                """
+                Compute magnitude-scaling term, equation (3), pag 1991.
+                """
+                # Computing Fm from equation (3), used in Fe. 
+                if rup.mag <= C['Mh']:
+                    return C['e0'] + \
+                        C['e1'] * (rup.mag - C['Mh']) + \
+                        C['e2'] * (rup.mag - C['Mh']) ** 2
+                else:
+                    return C['e0'] + \
+                        C['e1'] * (rup.mag - C['Mh'])
+
```

Comment:
I think this should be `C['e0'] + C['e3'] * (rup.mag - C['Mh'])` according to Eqn 3 of Yenier & Atkinson (2015) - this may be affecting the numbers you are getting